Content.Shared/Atmos/EntitySystems/SharedGasMinerSystem.cs

```diff
@@ -0,0 +1,65 @@
+using Content.Shared.Atmos.Components;
+using Content.Shared.Examine;
+using Content.Shared.Temperature;
+
+namespace Content.Shared.Atmos.EntitySystems;
+
+public abstract class SharedGasMinerSystem : EntitySystem
+{
+    [Dependency] private readonly ExamineSystemShared _examineSystemShared = default!;
+    [Dependency] private readonly SharedAtmosphereSystem _sharedAtmosphereSystem = default!;
+
+    // Gas miners are traditionally placed in a sealed room behind many layers of glass windows
+    // and empty space. Under normal conditions a player is usually at least 5 tiles from
+    // a gas miner. The standard examine range (3) is too small for the gas miner's examine text
+    // to be really useful, so we have an extended range for examining gas miners.
+    private const float MinerExamineRange = 6f;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<GasMinerComponent, ExaminedEvent>(OnExamine);
+    }
+
+    private void OnExamine(Entity<GasMinerComponent> ent, ref ExaminedEvent args)
+    {
+        var component = ent.Comp;
+
+        if (!_examineSystemShared.InRangeUnOccluded(args.Examiner, args.Examined, MinerExamineRange))
+            return;
+
```

Comment:
At this point I would just make it use the normal examine range (16 units) rather than special-casing and adding an extra raycast on top to be honest.

Reply:
I think when I was doing this my brain got the wires crossed between examine range and interaction range. Or maybe detail examine range? Anyway your totally right, I'll update it after I'm off work.